pkg/lifecycle/lifecycle.go

```diff
@@ -70,3 +70,27 @@ func (lc *Lifecycle) OnStop(ctx context.Context) error {
 	}
 	return g.Wait()
 }
+
+// OnStartSequentially runs models' Start function if models implmented it.
+func (lc *Lifecycle) OnStartSequentially(ctx context.Context) error {
+	for _, m := range lc.models {
+		if starter, ok := m.(Starter); ok {
+			if err := starter.Start(ctx); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+// OnStopSequentially runs models' Stop function if models implmented it.
+func (lc *Lifecycle) OnStopSequentially(ctx context.Context) error {
```

Comment:
it is best (and safer) to stop the services in reverse order, i.e., service start in order 1 --> 2 --> 3, then stop in order 3 --> 2 --> 1